### PR TITLE
Feature/mdx page

### PR DESCRIPTION
**Pull request by lebalz**
# MDX Page
A simple remark plugin that adds a `<MdxPage />` to every page that has a `page_id` in the frontmatter.

The `<MdxPage />` component has no visual representation, but enables two purposes:
1. initialize and load a RootDocument for each page, such that dynamic page content (such as comments) can be attached to this doc root. (The `page_id` is used as the id)
2. Since `<MdxPage />` is rendered as the first component, it's `useEffect`'s will be triggeret first - the perfect place to initialize e.g. the `Page` Meta Model

## Todo
- [x] parse the front-matter and only add the `<MdxPge />`  Compinent, when `page_id`  is present.

### CODE REVIEW

File: docs/Komponentengalerie/task-state.md
Comment (SilasBerger) [inline docs/Komponentengalerie/task-state.md:1]: 1. Should this section also mention the `enumerateAnswersPlugin`? As far as I can tell, this is required too, right?
2. Maybe add the same comment to `self-check.md`?

**Comment by SilasBerger** [comment]
LGTM, except for comment 👍

I see a lot of potential for this addition, such as access-restricted pages :)

**Review by SilasBerger — APPROVED** [review]

File: docs/Komponentengalerie/task-state.md
Comment (lebalz) [reply docs/Komponentengalerie/task-state.md:1]: Yes indeed, i added a docs page for the remark-enumerate-components plugin and linked it accordingly.

File: src/components/MdxPage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            pageStore.addIfNotPresent(pageId, true);
+        }
+        return () => {
+            pageStore.setCurrentPageId(undefined);
+        };
+    }, [pageId]);
+    React.useEffect(() => {
+        const { current } = pageStore;
+        if (current && userStore.viewedUserId) {
+            current.loadLinkedDocumentRoots();
```
Comment (SilasBerger) [inline src/components/MdxPage/index.tsx:30]: I believe this is the call that leads to the `403` issue for non-admin users mentioned in #17. As far as I can tell, this leads to a call to `/documents` (API: `router.ts#88`), which is restricted to admins (API: `documentRoots.ts#62-64`). If I disable this line, the error goes away.

Comment (lebalz) [reply src/components/MdxPage/index.tsx:30]: Yes exactly: https://github.com/GBSL-Informatik/teaching-dev/pull/17#issuecomment-2401666399 (i was a bit too late ;))

**Pull request merged**